hackvote/app/controllers/events_controller.rb

```diff
@@ -0,0 +1,42 @@
+class EventsController < ApplicationController
+	include ApplicationHelper
+
+	def new
+		@event = Event.new
+		@cur_events = Event.where( completed: false)
+		@past_events = Event.where( completed: true)
+	end
+
+	def create
+		@event = Event.new(event_params)
+		if @event.save
+			cookies.permanent[@event.id] = 3
+			redirect_to @event
+		else
+			render 'new'
+		end
+	end
+
+	def show
+		@event = Event.find(params[:id])
```

Comment:
find causes an exception on failure. find_by_id is better to use.

and you should do params[:id].to_i otherwise you are doing a string look up on an integer which is inefficient.